__tests__/fetchers.js

```diff
@@ -433,3 +433,78 @@ test('TarballFetcher.fetch throws on truncated tar header', async () => {
     new RegExp(reporter.lang('errorExtractingTarball', '.*', '.*broken-tar-header.tgz')),
   );
 });
+
+test('TarballFetcher.fetch retries on a truncated response', async () => {
+  const dir = await mkdir('tarball-fetcher');
+  const config = await Config.create({});
+
+  const workingUrl = 'https://registry.npmjs.org/lodash.isempty/-/lodash.isempty-4.4.0.tgz';
+  // This one returns a truncated response - body is shorter than the Content-Length header.
+  const brokenUrl = 'https://registry.npmjs.org/lodash.isempty/-/lodash.isempty-4.4.0-broken.tgz';
+
+  const mockRequest = (require: any).requireActual('./__mocks__/request');
+
+  let retryCount = 2;
+
+  // Shorter retry delay for tests
+  config.requestManager.retryDelay = 10;
```

Comment:
Why not outright 0?

Reply:
Can make it 0 - shouldn't have a significant impact either way.